airflow/providers/samba/hooks/samba.py

```diff
@@ -31,52 +62,202 @@ class SambaHook(BaseHook):
     conn_type = 'samba'
     hook_name = 'Samba'
 
-    def __init__(self, samba_conn_id: str = default_conn_name) -> None:
+    def __init__(self, samba_conn_id: str = default_conn_name, share: Optional[str] = None) -> None:
         super().__init__()
-        self.conn = self.get_connection(samba_conn_id)
+        conn = self.get_connection(samba_conn_id)
+
+        if not conn.login:
+            LOGGER.info("Login not provided")
+
+        if not conn.password:
+            LOGGER.info("Password not provided")
+
+        self._host = conn.host
+        self._share = share or conn.schema
+        self._connection_cache = connection_cache = {}
+        self._conn_kwargs = {
+            "username": conn.login,
+            "password": conn.password,
+            "port": conn.port or 445,
+            "connection_cache": connection_cache,
```

Comment:
Should this be `self._connection_cache`?

Reply:
Well, the variable is aliased locally to `connection_cache`.

Reply:
Oh OK, I misread.